lib/pure/concepts.nim

```diff
@@ -0,0 +1,35 @@
+#
+#
+#            Nim's Runtime Library
+#        (c) Copyright 2020 Andreas Rumpf
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+
+## The ``concepts`` module contains implementations of standard concepts used
+## across Nim's standard library.
+
+type
+  Iterable*[T] = concept x
```

Comment:
we now have `iterable[T]` which allows matching called iterators, unlike this `Iterable` concept, eg:
```nim
iterator iota(n: int): int =
  for i in 0..<n: yield i
proc sum[T](a: iterable[T]): T =
  for ai in a: result += ai
assert 3.iota.sum == 0+1+2
```